projects/templates/src/lib/components/po-page-dynamic-detail/po-page-dynamic-detail.component.ts

```diff
@@ -213,34 +315,35 @@ export class PoPageDynamicDetailComponent implements OnInit {
     return util.valuesFromObject(keys).join('|');
   }
 
-  private goBack(/*path*/) {
+  private goBack() {
     window.history.back();
-    // if (path) {
-    //   this.navigateTo({ path, component: PoPageDynamicEditComponent });
-    // } else {
-    //   window.history.back();
-    // }
   }
 
   private loadData(id) {
-    this.poPageDynamicService.getResource(id).toPromise().then(response => {
-      this.model = response;
-    }).catch(() => {
-      this.model = undefined;
-      this.actions = undefined;
-    });
+    return this.poPageDynamicService.getResource(id).pipe(
+      tap(response => this.model = response),
+      catchError(error => {
+        this.model = undefined;
+        this.actions = undefined;
+        return throwError(error);
+      })
+    );
   }
 
-  private loadMetadata(id) {
-    this.poPageDynamicService.getMetadata<any>('detail').toPromise().then(response => {
-      this.autoRouter = response.autoRouter;
-      this.actions = response.actions || {};
-      this.breadcrumb = response.breadcrumb || { items : [] };
-      this.fields = response.fields || [];
-      this.title = response.title;
+  private getMetadata(serviceApi: string): Observable<PoPageDynamicDetailMetaData> {
+    if (serviceApi) {
```

Comment:
Sugestão:
alterar `serviceApi` para `serviceApiFromRoute`

```suggestion
  private getMetadata(serviceApiFromRoute: string): Observable<PoPageDynamicDetailMetaData> {
    if (serviceApiFromRoute) {
```

Reply:
Feito